store/router.go

```diff
@@ -42,112 +52,192 @@ func (r *Router) Get(ctx context.Context, key []byte, cm commitments.CommitmentM
 			return nil, err
 		}
 
-		if actualHash := crypto.Keccak256(value); !utils.EqualSlices(actualHash, key) {
-			return nil, fmt.Errorf("expected key %s to be the hash of value %s, but got %s",
-				hexutil.Encode(key), hexutil.Encode(value), hexutil.Encode(actualHash))
+		err = r.s3.Verify(key, value)
+		if err != nil {
+			return nil, err
 		}
-
 		return value, nil
 
 	case commitments.SimpleCommitmentMode, commitments.OptimismAltDA:
-		if r.mem != nil {
-			r.log.Debug("Retrieving data from memstore")
-			return r.mem.Get(ctx, key)
+		if r.eigenda == nil {
+			return nil, errors.New("expected EigenDA backend for DA commitment type, but none configured")
 		}
+
+		// 1 - read blob from cache if enabled
+		if r.cacheEnabled() {
+			r.log.Debug("Retrieving data from cached backends")
+			data, err := r.multiSourceRead(ctx, key, false)
+			if err == nil {
+				return data, nil
+			}
+			r.log.Warn("Failed to read from cache targets", "err", err)
+		}
+
+		// 2 - read blob from EigenDA
```

Comment:
The flow and all handling of `Get` makes sense to me and seems logically intact, but I feel like this flow can be organized more clearly 🤔 like, `multiSourceRead` could take in the source targets directly instead of having a boolean flag, and wrap around `eigenda.Get` and `.Verify` as well

Reply:
true but you'd have to overload the caller with mutex mgmt - gonna leave as if ok. don't see how this approach would present cleaner abstraction.

Reply:
sounds good!